From f8dc1b1a581d8ab6f78b3560bce7f8db8c3e8a0c Mon Sep 17 00:00:00 2001
From: Zhou Peng <eagle.zhou@nxp.com>
Date: Mon, 14 May 2018 20:21:44 +0800
Subject: [PATCH 3795/5242] MLK-18301-3 - [i.MX8MM/Hantro]: Enable hantro vpu
 on mscale 845S platform

commit  7dd27dd857a6fdef857b04ab05003750278499c8 from
https://source.codeaurora.org/external/imx/linux-imx.git

Enable hantro 845 decoder/encoder on device tree

Signed-off-by: Zhou Peng <eagle.zhou@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8mm-evk.dts |   12 ++++++++++++
 arch/arm64/boot/dts/freescale/fsl-imx8mm.dtsi    |   17 ++++++++---------
 drivers/mxc/Kconfig                              |    2 ++
 drivers/mxc/Makefile                             |    2 ++
 4 files changed, 24 insertions(+), 9 deletions(-)
 mode change 100644 => 100755 arch/arm64/boot/dts/freescale/fsl-imx8mm-evk.dts
 mode change 100644 => 100755 arch/arm64/boot/dts/freescale/fsl-imx8mm.dtsi

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk.dts
old mode 100644
new mode 100755
index 821f61e..a9726e2
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk.dts
@@ -537,3 +537,15 @@
 &gpu {
 	status = "okay";
 };
+
+&vpu_g1 {
+	status = "okay";
+};
+
+&vpu_g2 {
+	status = "okay";
+};
+
+&vpu_h1 {
+	status = "okay";
+};
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mm.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8mm.dtsi
old mode 100644
new mode 100755
index 20b920e..74eb52c
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mm.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mm.dtsi
@@ -851,22 +851,22 @@
 	};
 
 	vpu_h1: vpu_h1@38320000 {
-		compatible = "nxp,imx8mq-hantro-h1";
+		compatible = "nxp,imx8mm-hantro-h1";
 		reg = <0x0 0x38320000 0x0 0x10000>;
 		reg-names = "regs_hantro_h1";
 		interrupts = <GIC_SPI 30 IRQ_TYPE_LEVEL_HIGH>;
 		interrupt-names = "irq_hantro_h1";
-		clocks = <&clk IMX8MM_CLK_VPU_H1_ROOT>;
-		clock-names = "clk_hantro_h1";
+		clocks = <&clk IMX8MM_CLK_VPU_H1_ROOT>, <&clk IMX8MM_CLK_VPU_DEC_ROOT>;
+		clock-names = "clk_hantro_h1", "clk_hantro_h1_bus";
+		assigned-clocks = <&clk IMX8MM_CLK_VPU_H1_SRC>,<&clk IMX8MM_CLK_VPU_BUS_SRC>;
+		assigned-clock-parents = <&clk IMX8MM_VPU_PLL_OUT>, <&clk IMX8MM_SYS_PLL1_800M>;
+		assigned-clock-rates = <600000000>, <800000000>;
 		power-domains = <&vpu_h1_pd>;
-		assigned-clocks = <&clk IMX8MM_CLK_VPU_H1_SRC>;
-		assigned-clock-parents = <&clk IMX8MM_VPU_PLL_OUT>;
-		assigned-clock-rates = <600000000>;
 		status = "disabled";
 	};
 
 	vpu_g1: vpu_g1@38300000 {
-		compatible = "nxp,imx8mq-hantro";
+		compatible = "nxp,imx8mm-hantro";
 		reg = <0x0 0x38300000 0x0 0x100000>;
 		reg-names = "regs_hantro";
 		interrupts = <GIC_SPI 7 IRQ_TYPE_LEVEL_HIGH>;
@@ -881,7 +881,7 @@
 	};
 
 	vpu_g2: vpu_g2@38310000 {
-		compatible = "nxp,imx8mq-hantro";
+		compatible = "nxp,imx8mm-hantro";
 		reg = <0x0 0x38310000 0x0 0x100000>;
 		reg-names = "regs_hantro";
 		interrupts = <GIC_SPI 8 IRQ_TYPE_LEVEL_HIGH>;
@@ -894,7 +894,6 @@
 		power-domains = <&vpu_g2_pd>;
 		status = "disabled";
 	};
-
 	gpu: gpu@38000000 {
 		compatible ="fsl,imx6q-gpu";
 		reg = <0x0 0x38000000 0x0 0x8000>, <0x0 0x38008000 0x0 0x8000>,
diff --git a/drivers/mxc/Kconfig b/drivers/mxc/Kconfig
index 9745723..e0b9d68 100755
--- a/drivers/mxc/Kconfig
+++ b/drivers/mxc/Kconfig
@@ -19,6 +19,8 @@ config MXC_SIM
 source "drivers/mxc/sim/Kconfig"
 
 if ARCH_MXC_ARM64
+source "drivers/mxc/hantro_845/Kconfig"
+source "drivers/mxc/hantro_845_h1/Kconfig"
 source "drivers/mxc/vpu-malone/Kconfig"
 source "drivers/mxc/vpu-decoder-b0/Kconfig"
 source "drivers/mxc/vpu-encoder-b0/Kconfig"
diff --git a/drivers/mxc/Makefile b/drivers/mxc/Makefile
index 302c309..c6bcc9f 100755
--- a/drivers/mxc/Makefile
+++ b/drivers/mxc/Makefile
@@ -6,6 +6,8 @@ obj-$(CONFIG_MXC_HDMI_CEC) += hdmi-cec/
 obj-$(CONFIG_MXC_GPU_VIV) += gpu-viv/
 obj-$(CONFIG_MXC_MIPI_CSI2) += mipi/
 obj-$(CONFIG_MXC_HANTRO) += hantro/
+obj-$(CONFIG_MXC_HANTRO_845) += hantro_845/
+obj-$(CONFIG_MXC_HANTRO_845_H1) += hantro_845_h1/
 obj-$(CONFIG_MXC_VPU_MALONE) += vpu-malone/
 obj-$(CONFIG_MX8_HDP)	+= hdp/
 obj-$(CONFIG_MXC_VPU_DECODER) += vpu-decoder-b0/
-- 
1.7.9.5

